src/amplitude-client.js

```diff
@@ -79,6 +79,12 @@ AmplitudeClient.prototype.init = function init(apiKey, opt_userId, opt_config, o
     this.options.apiKey = apiKey;
     this._storageSuffix = '_' + apiKey + this._legacyStorageSuffix;
 
+    var hasExistingCookie = !!this.cookieStorage.get(this.options.cookieName + this._storageSuffix);
```

Comment:
moved into the try catch to ensure that `this._storageSuffix` is defined

Reply:
i kind of like that you're using the cookie to signal if the user has accepted tracking

Reply:
i feel like that's the safest bet to carry over. i haven't fully thought of the case where our cookie expires tho.... 😬